src/cryptonote_core/loki_name_system.cpp

```diff
@@ -220,34 +219,25 @@ static bool sql_compile_statement(sqlite3 *db, char const *query, int query_len,
   return result;
 }
 
-burn_type mapping_type_to_burn_type(mapping_type in)
-{
-  burn_type result = burn_type::custom;
-  switch (in)
-  {
-    case mapping_type::lokinet: result = burn_type::lokinet_1year; break;
-    case mapping_type::session: result = burn_type::session; break;
-    case mapping_type::wallet: result = burn_type::wallet; break;
-    default: break;
-  }
-  return result;
-}
-
-uint64_t burn_requirement_in_atomic_loki(uint8_t /*hf_version*/, burn_type type)
+uint64_t burn_requirement_in_atomic_loki(uint8_t /*hf_version*/, mapping_type type)
 {
   uint64_t result = 0;
   switch (type)
   {
-    case burn_type::update_record:
+    case mapping_type::update_record_internal:
       result = 0;
       break;
 
-    case burn_type::lokinet_1year: /* FALLTHRU */
-    case burn_type::session: /* FALLTHRU */
-    case burn_type::wallet: /* FALLTHRU */
-    case burn_type::custom: /* FALLTHRU */
-    default: result = 30 * COIN;
+    case mapping_type::lokinet_1year: /* FALLTHRU */
+    case mapping_type::session: /* FALLTHRU */
+    case mapping_type::wallet: /* FALLTHRU */
+    default:
+      result = 20 * COIN;
       break;
+
+    case mapping_type::lokinet_2years: result = 40 * COIN; break;
+    case mapping_type::lokinet_5years: result = 80 * COIN; break;
+    case mapping_type::lokinet_10years: result = 120 * COIN; break;
```

Comment:
I've moved blink and lns burns into loki_economy since they are a simple copy and paste in. Staking requirement and block reward not yet because they want loki::exp2/round2.